[peripherals] fix screensaver shot away

AlwinEsch opened the pull request:
## Description
<!--- Describe your change in detail -->

The Joystick monitor triggers always a "OnAxisMotion(...)" also without a
real axis change.

This fix the always direct cone away mouse pointer after mouse move and the
screensavers who are startet and immediately stopped.

@garbear can you check this is OK
<!--- Provide a general summary of your change in the Title above -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch commented on xbmc/input/joysticks/JoystickMonitor.cpp:
```diff
@@ -22,6 +22,8 @@
 #include "Application.h"
 #include "input/InputManager.h"
 
+#define AXIS_THRESHOLD       0.75f // Axis must exceed this value to be mapped
```
This is taken from ./xbmc/input/joysticks/generic/ButtonMapping.cpp,  makes it sense to define in header to have global?

garbear replied:
No, the threshold is application-dependent. For example, I chose 0.75f in the button mapper because pressure-sensitive buttons send both an analog and a digital button press, and this gives more weight to the digital button press. Here, this threshold determines the sensitivity to awakening the screen saver. 0.75f is a good value, though I don't see any reason it shouldn't be 0.5f.

garbear commented:
on a snowboarding trip, I can review when I have some down time

AlwinEsch commented:
@garbear still on snowboarding trip? :-D ;-)

The problem is still present in Kodi, what do you think about them here?

garbear commented on xbmc/input/joysticks/JoystickMonitor.cpp:
```diff
@@ -49,7 +51,9 @@ bool CJoystickMonitor::OnHatMotion(unsigned int hatIndex, HAT_STATE state)
 bool CJoystickMonitor::OnAxisMotion(unsigned int axisIndex, float position, int center, unsigned int range)
 {
-  if (position)
+  const bool bIsActive = (std::abs(position) >= AXIS_THRESHOLD);
```
This calculation needs to change since the Peripheral API bump to 1.3.0. See https://github.com/xbmc/xbmc/blob/c80e266/xbmc/input/joysticks/generic/InputHandling.cpp#L69 for example.

This should be:

```c++
const float magnitude = std::abs(position - center) / range;

if (magnitude >= AXIS_THRESHOLD)
{
  ...
```

garbear commented:
Unfortunately the solution is more complicated than this. Accelerometers and unmapped buttons will send erratic positions, and we also don't know the true center and range. This class should only interrupt the screen saver if the axis is mapped to a button on the controller.

AlwinEsch commented:
You are right there is a bit more needed, I close this them.

garbear commented:
Is this still an issue, or has it been fixed in one of my many joystick PRs?

AlwinEsch commented:
Was more a system fault and fixed here with https://askubuntu.com/questions/585706/deactivate-accelerometer

With them active are also not working games.

There was a fix from @wsnipex https://github.com/xbmc/xbmc/commit/d8d6f0842a0ef1f325b94d7c31d9ff3829905732 but was gone on other changes

garbear commented:
What if we only allow mapped axes to wake from screen saver?

AlwinEsch committed:
[peripherals] fix screensaver shot away

The Joystick monitor triggers always a "OnAxisMotion(...)" also without a
real axis change.

This fix the always direct cone away mouse pointer after mouse move and the
screensavers who are startet and immediately stopped.

AchimTuran commented:
@AlwinEsch ist this issue still present or should we close this PR?

Rechi commented:
@AlwinEsch this needs a rebase

Rechi commented:
@garbear is this obsolete with #12892?

garbear commented:
Yes
